### PR TITLE
Feature/Google Maps

### PR DESCRIPTION
**Pull request by priyankj-simform**
Add - Google Maps compose ground overlay, tile overlay, indoor map, lite map, and map in scroll view examples.

The image names are changed to meaning full name that's why extra files are changed. The images are moved to xxhdpi from v24 as the minimum SDK version is 21 and these images will generate exception if opened on the lower android version.

Change in DataProvider class's package so that it will reflect all activities.

### CODE REVIEW

File: app/src/main/java/com/jetpack/compose/learning/maps/indoorlevel/MapIndoorActivity.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    fun MapsExample(
+        cameraPositionState: CameraPositionState,
+        onMapLoaded: () -> Unit
+    ) {
+        GoogleMap(
+            modifier = Modifier.fillMaxSize(),
+            cameraPositionState,
+            onMapLoaded = onMapLoaded,
+            properties = MapProperties(isIndoorEnabled = true),
+            indoorStateChangeListener = object : IndoorStateChangeListener {
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/maps/indoorlevel/MapIndoorActivity.kt:116]: Where this listener has been used?

File: app/src/main/java/com/jetpack/compose/learning/maps/indoorlevel/MapIndoorActivity.kt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                override fun onIndoorBuildingFocused() {
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/maps/indoorlevel/MapIndoorActivity.kt:117]: Seems Redundant overriding method

File: app/src/main/java/com/jetpack/compose/learning/maps/indoorlevel/MapIndoorActivity.kt [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                    super.onIndoorBuildingFocused()
+                    //Called on indoor building is focused
+                }
+
+                override fun onIndoorLevelActivated(building: IndoorBuilding) {
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/maps/indoorlevel/MapIndoorActivity.kt:122]: Seems Redundant overriding method

File: app/src/main/java/com/jetpack/compose/learning/maps/overlay/MapGroundOverlayViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import androidx.lifecycle.ViewModel
+import com.jetpack.compose.learning.R
+import com.jetpack.compose.learning.maps.GroundOverlayMapUIState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class MapGroundOverlayViewModel : ViewModel() {
+    private val imageList = listOf(
+        R.drawable.ic_statue_of_liberty,
+        R.drawable.wp1,
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/maps/overlay/MapGroundOverlayViewModel.kt:12]: Image/resource name should be meaningful.

File: app/src/main/java/com/jetpack/compose/learning/maps/overlay/MapGroundOverlayViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    private val _state = MutableStateFlow(GroundOverlayMapUIState())
+    val state: StateFlow<GroundOverlayMapUIState>
+        get() = _state
+
+    fun changeImage() {
+        val randomImage = imageList.random()
+        if (randomImage != _state.value.imageResource) {
+            _state.value = _state.value.copy(imageResource = randomImage)
+        } else {
+            changeImage()
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/maps/overlay/MapGroundOverlayViewModel.kt:29]: Seems method invocation will be infinite. It is calling itself in the else condition.

File: app/src/main/java/com/jetpack/compose/learning/maps/basic/LiteMapInListActivity.kt [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+            ) {
+                Marker(state = MarkerState(place.position), title = place.title)
+            }
+        }
+    }
+}
+
+data class TouristPlace(val id: Int, val title: String, val position: LatLng)
+
+private fun getMockTouristPlace(): List<TouristPlace> {
```
Comment (mobile-simformsolutions) [inline app/src/main/java/com/jetpack/compose/learning/maps/basic/LiteMapInListActivity.kt:117]: Add this to model class or view model

Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/maps/basic/LiteMapInListActivity.kt:117]: Add this to model class or view model

File: app/src/main/java/com/jetpack/compose/learning/maps/GoogleMapsStates.kt [hunk 7]
```diff
@@ -158,4 +163,26 @@ data class CircleMapUIState(
...
+data class GroundOverlayMapUIState(
+    @DrawableRes val imageResource: Int,
+    val bearing: Float = 0f,
+    val transparency: Float = 0f,
+    val clickable: Boolean = false,
+    val visible: Boolean = true
+) {
+    fun getImageBitmap(resource: Resources): BitmapDescriptor {
+        val bitmap = BitmapFactory.decodeResource(resource, imageResource)
+        val create = RoundedBitmapDrawableFactory.create(resource, bitmap)
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/maps/GoogleMapsStates.kt:177]: keep meaningful name

File: app/src/main/java/com/jetpack/compose/learning/maps/indoorlevel/MapIndoorActivity.kt [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+import com.jetpack.compose.learning.theme.AppThemeState
+import com.jetpack.compose.learning.theme.BaseView
+import com.jetpack.compose.learning.theme.SystemUiController
+
+/**
+ * Indoor level example.
+ * Indoor level represents a level in a building.
+ */
+class MapIndoorActivity : ComponentActivity() {
+    private val madisonSquareGarden = LatLng(40.7504861, -73.9935901)
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/maps/indoorlevel/MapIndoorActivity.kt:49]: change object name

File: app/src/main/java/com/jetpack/compose/learning/maps/basic/LiteMapInListActivity.kt (hunk 6 above)
Comment (yashwantgowla-simform) [reply app/src/main/java/com/jetpack/compose/learning/maps/basic/LiteMapInListActivity.kt:117]: Or let's make a class for DataProvider for all the other activities

**Review by yashwantgowla-simform — APPROVED** [review]

**Pull request merged**